### ISSUE TITLE
Docker: Support using name and count together

### ISSUE THREAD

**Issue by Endron** [ansible/ansible-modules-core#1146]
As stated in the documentation it is currently not possible to use the options "name" and "count" of the Docker module in combination. This is limiting. I need to use name to have a unique identifier to be able to use ansible to replace a started container with a new image:

```
tasks:
  - name: deploy jenkins
    docker: 
      name: "jenkins"
      image: "jenkins:1.593" # <-- Change image here
      detach: true
      pull: always
      state: reloaded
```

This works for container that are only ever deployed once.
But not being able to give a count (other than 1) limits usability when working with micro services: In this case I am likely to have multiple instances of each service deployed and I want to use ansible to change the number of deployed services.

I could define each service as its own task but this will not scale very well and would probably make the playbook hard to read. Just setting an appropriate count would yield a much clearer playbook.

This could properly be implemented by using a template for container names. All container would be named service_1 to service_n so ansible can figure out they belong together. This could be improved by using customizable templates.

Customizable templates would open up even more usecases as users might be able to do interesting things by stopping containers that happen to match with the template.

**Comment by wangyun** [comment]
+1

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@Endron We appreciate you taking the time to file this issue, but the module you've opened the issue about has been deprecated.

We will consider critical bugfixes for deprecated modules, but all other types of requests will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
